Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -1020,6 +1024,19 @@ private extension AppSettingsStore {
         let timeRange = StatsTimeRangeV4(rawValue: timeRangeRawValue)
         onCompletion(timeRange)
     }
+
+    func setLastSelectedMostActiveCouponsTimeRange(siteID: Int64, timeRange: MostActiveCouponsTimeRange) {
+        let storeSettings = getStoreSettings(for: siteID)
+        let updatedSettings = storeSettings.copy(lastSelectedMostActiveCouponsTimeRange: timeRange.rawValue)
+        setStoreSettings(settings: updatedSettings, for: siteID)
+    }
+
+    func loadLastSelectedMostActiveCouponsTimeRange(siteID: Int64, onCompletion: (MostActiveCouponsTimeRange?) -> Void) {
+        let storeSettings = getStoreSettings(for: siteID)
+        let timeRangeRawValue = storeSettings.lastSelectedMostActiveCouponsTimeRange
+        let timeRange = MostActiveCouponsTimeRange(rawValue: timeRangeRawValue)
+        onCompletion(timeRange)
+    }
```

Comment:
Unit tests will be added as part of https://github.com/woocommerce/woocommerce-ios/issues/12728